### PR TITLE
Feature: casl permissions for tech

### PR DESCRIPTION
**Pull request by Ajen07**
# Description

- Add new permissions for Tech endpoint

## Issue link

[Fixes # (issue)](https://app.clickup.com/t/86azr0q42)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Feature updates / changes
- [ ] Tests
- [] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- Test with  `yarn test:docker`

- Tested with new test cases for the above change

-  Tested in swagger 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the change log

### CODE REVIEW

**Review by cherylli — CHANGES_REQUESTED** [review]
It's good that you've added some new tests for 403, but I think we should also update the endpoint annotations in the controller file (i.e. swagger)

other things look good to me

**Comment by Ajen07** [comment]
> It's good that you've added some new tests for 403, but I think we should also update the endpoint annotations in the controller file (i.e. swagger)
> 
> other things look good to me

what changes in annotations should I make ?

File: test/techs.e2e-spec.ts [hunk 1]
```diff
@@ -104,6 +104,38 @@ describe("Techs Controller (e2e)", () => {
                     );
                 });
         });
+        it("should return 404 if invalid teamId provided", async () => {
```
Comment (cherylli) [inline test/techs.e2e-spec.ts:107]: re: [your comment](https://github.com/chingu-x/chingu-dashboard-be/pull/174#issuecomment-2213376203)

for example here, you added 401 and 403 but I noticed it's not already on swagger. This particular endpoint only has 200 under responses

image

Comment (Ajen07) [reply test/techs.e2e-spec.ts:107]: Okay I will add these responses.

**Review by cherylli — APPROVED** [review]
thanks!

**Review by JoshuaHinman — APPROVED** [review]
All tests passed, and swagger returned the expected responses for user not_in_voyage - nice work

**Pull request merged**